transfer_nlp/runner/experiment_runner.py

```diff
@@ -141,3 +142,5 @@ def run_all(experiment: Union[str, Path, Dict],
                 reporter.report(exp_name, experiment_config, exp_report_path)
             finally:
                 ExperimentRunner._stop_log_capture(log_handler)
+                
+        return experiment_config_cache
```

Comment:
should we return the experiment config cache, or change the `ReporterABC.report`  to return `Any` and then return a dictionary of exp name --> report data? 

another option instead of returning the value would be to add another method to the ReporterABC called reportAll that would take the dictionary as a parameter and a report directory. 

also people can currently work around this issue by using global variables, but i agree it's not pretty...

Reply:
I like the first suggestion!